### PR TITLE
fix(hc): Introduce OrganizationService.get_default_organization

### PR DESCRIPTION
**Pull request by RyanSkonnord**
Fix a silo access error when running a typical dev environment in the single-org and split silo modes. Introduce a special region resolution strategy that requires there to be only one region, which is a safe assumption when there is only one organization.

### CODE REVIEW

**Comment by RyanSkonnord** [comment]
This is working as expected when observed in a dev environment, but it's in draft status because the new `RequireSingleOrganization` isn't printing debug statements as I expect. I'm going to see if `frontend/auth_login.py` needs any additional unit test coverage to exercise that path, and will dig in with a debugger to make sure it's working as it should.

Other than that, this should completely unblock the auth flow in the split silo dev environment!

**Review by corps — APPROVED** [review]
Nice find!

**Comment by dashed** [comment]
Looks good to me! Just need some tests 👍

**Comment by RyanSkonnord** [comment]
> By the way, if no regions can be found, what about using `SENTRY_MONOLITH_REGION`? Check the `GlobalRegionDirectory` constructor logic.
> 
> It might be worth converging some of this logic.

Like this? 69ba2ef689

Would that do anything? If the default org exists, we should be able to count on its region existing. I guess it could lead to  a more graceful failure mode if the server is in such an early state that even the default org hasn't been created yet. That overlaps with https://github.com/getsentry/sentry/pull/52537 and https://github.com/getsentry/sentry/pull/52560 -- i.e., we're going to fail to find the actual org in the next step, but we avoid crashing on the region resolution step.

**Review by dashed — APPROVED** [review]

File: src/sentry/services/hybrid_cloud/organization/impl.py [hunk 1]
```diff
@@ -110,6 +110,9 @@ def get_org_by_slug(
         except Organization.DoesNotExist:
             return None
 
+    def get_default_region(self):
```
[the remark this answers is not in the document]
Comment (RyanSkonnord) [reply src/sentry/services/hybrid_cloud/organization/impl.py:113]: Oops, what a weird mistake. I must have been tunnel-visioned on the region resolution problem when I typed that. Thanks.

**Pull request merged**